resources/authgear/templates/en/web/authflowv2/ldap_login.html

```diff
@@ -64,8 +92,7 @@ <h1 class="screen-title">
       "Type" "old-password"
       "AutoFocus" $.ShouldFocusInput
       "Classname" "w-full"
-      "HasError" $.PasswordInputError.HasError
-      "HasErrorMessage" $.PasswordInputError.HasErrorMessage
+      "Error" $pw_err
```

Comment:
I take a closer at this, and I found after your change, the behavior is different. Have you ever tried the original behavior?

Reply:
Right, missed that `InvalidCredentials` should be 

1. `both username and password input` red border
2. no error messages for input fields
3.  show alert message by `Invalid credentials`

Lemme fix that 🙏

Reply:
So as I said, you do not need to change the interface of __password_field.html

Reply:
>  do not need to change the interface of __password_field.html
Disagree on this. I think we need `.Error` to be passed into `__password_field.html` for below

```
<p class="input__error-message" data-text-field-target="errorMessage">
      {{ template "authflowv2/__error.html" (merge (dict "Error" $.Error) $.Ctx) }}
    </p>
```

Reply:
So I suggest below diff

```diff
- "HasError`
+ "Error"
   "HasErrorMessage" // keep

```

Reply:
OK. Remove .HasErrorMessage and add .ErrorMessage

Then in __password_field.html, render .ErrorMessage directly.

Reply:
I want it to support two things

- A flag (or a prop) to make it in red
- A prop to pass it an error message to render

So this becomes

- HasError
- ErrorMessage

Reply:
Sounds great. Lemme implement that.

Will do similarly for `__new_password_field.html` 🙏 Thanks daic

Reply:
Sor @louischan-oursky , on second thought, why not pass the entire Error over HasError flag?

So it becomes

- Error
- ErrorMessage

Reply:
> Have you ever tried the original behavior?

Did you?

If you have, then you will see in a case, the field is just in red, without any error message.

Reply:
> Did you?

Yes I did. I understand that we want 2 separate controls on `field red` and `has error message`

I am suggesting `Error` over `HasError` because that way the consumer `__password_field.html` has control over the entire Error object, instead of a processed flag.

It's rather trivial, but just wanna point it out

Reply:
The rationale for __password_field.html to consume an entire Error object is __password_field.html needs to do something with .Error. But it does not need to. It just need to display an given error message.

Or you found such a case?

Reply:
> Or you found such a case?

Nope. Maybe let's just pass it when it's needed (which is unlikely).

Will implement below interface 

- HasError
- ErrorMessage

for new_password_field, will be

```
"HasNewPasswordError" $has_pw_err
    "NewPasswordErrorMessage" $pw_error_message
    "HasConfirmPasswordError" $confirm_pw_err
    "ConfirmPasswordErrorMessage" $confirm_pw_error_message
```